person.rb

```diff
@@ -20,6 +22,16 @@ def correct_name
     @name
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.students << self
+  end
+
+  def add_rental(rental)
```

Comment:
---
Hello,

Great job on working through the relationships between `Book`, `Person`, and `Rental`. I noticed a few areas where we need to align your implementation with the project requirements to ensure the many-to-many relationship is properly set up.

1. **`add_rental` Method in `Book` and `Person` Classes:**
   Currently, your `add_rental` methods in both `Book` and `Person` classes expect a `Rental` instance as a parameter. However, according to the project requirements, these methods should be responsible for creating the `Rental` instance. This ensures encapsulation and manages the associations within the classes. Here's the suggested adjustment:

   In `Book`:
   ```ruby
   def add_rental(person, date)
     @rentals << Rental.new(date, self, person)
   end
   ```

   In `Person`:
   ```ruby
   def add_rental(book, date)
     @rentals << Rental.new(date, book, self)
   end
   ```

   By doing this, each class takes responsibility for creating `Rental` objects and establishing the "has-many" side of the relationship.



By implementing these adjustments, you'll have a strong many-to-many relationship setup that follows the project requirements and encapsulates the responsibility of managing associations within each class.

Keep up the good work and let me know if you have any questions.

Best,
Temitope

---